lib/entry/crossDCsStreamPersistence.js

```diff
@@ -0,0 +1,191 @@
+var glob = require('glob');
+var path = require('path');
+var backendClient = require('../BackendClientFactory.js').getBackendClient();
+var PlaylistGenerator = require('./../playlistGenerator/PlaylistGenerator');
+const endListPendingEntriesFileName = "endListPending.json";
+const stoppedEntriesPath = config.get('stoppedEntriesPath');
+
+// todo:
+// 1) update the sequence of calling BE to verify whether EntryServerNodes > 0
+// 2) when no DC serves the stream do:
+//   2.1) update stoppedState to ended. ---> using this.playlistGenerator.
+//   2.2) save playlist. ---> using this.playlistGenerator
+//   2.3) remove from list
+// 3) add sequence to restore _entriesToProcess, when LiveController starts!!!
+
+class CrossDCsStreamPersistence {
+
+    constructor() {
+ 
+        this.stoppedEntriesFileName = path.join(stoppedEntriesPath, endListPendingEntriesFileName)
+        this.checkStreamIntervalMSec = config.get('streamPersistenceTimerMSec');
+        this._entriesToProcess = {};
+        this._entriesData = {};
+    }
+}
+
+
+//---------------------------------------------------------------
+// Todo: Lilach: PLAT-5023
+// 1) check if any DC provides the entryId:
+//  1.1) in case no DC provides the entry: update the time tag +
+//  1.2) otherwise,
+//      1.2.1) start timer to check if entry is provided
+//      1.2.2) rename playlist.json
+//--------------------------------------------------------------
+CrossDCsStreamPersistence.prototype.streamPersistenceCheck = function (entryId) {
+
+    var deferred = Q.defer();
+
+    backendClient.getLiveEntryServerNodes(entryId)
+        .then((liveEntryServerNodes) => {
+            if (liveEntryServerNodes.count > 0) {
+                this._entriesData[entryId].timestamp = Date.now;
+                this._entriesData[entryId].playlist.persistStopped();
+            } else {
+                this._entriesData[entryId].playlist.end();
+                delete this._entriesData[entryId].playlist;
+                delete this._entriesToProcess[entryId];
+            }
+            deferred.resolve(count);
+        })
+        .catch((err) => {
+            logger.error(`failed to get entry server nodes for stopped entry [${entryId}], error: [${ErrorUtils.error2string(err)}]`);
+            deferred.reject(err);
+        });
+
+    return deferred.promise;
+
+};
+
+CrossDCsStreamPersistence.prototype.addEntry = function addEntry(playlistGenerator) {
+    var that = this;
+    let entryId = playlistGenerator.entryId;
+
+    return Q.fcall(function () {
+        if (that._entriesToProcess[entryId]) {
+            logger.info("[%s] entry is already in stream persistence list, ignoring request", entryId);
+            // todo: replace the playlist obj (consult with Guy)
+            // return Q.resolve(); required ???
+            return true;
+        } else {
+            // keep the playlistGenerator obj
+            this._entriesData[entryId] = { playlist: playlistGenerator, timestamp: new Date() };
```

Comment:
why not keep the only the playlistgenerator? the timestamp is already kept inside the playlistgenerate, just expose it